client/src/controllers/SyncController.ts

```diff
@@ -90,6 +90,59 @@ export class SyncController extends Controller<HTMLInputElement> {
     });
   }
 
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  prefillTitleFromFilename(_event?: Event) {
+    const input = this.element;
+
+    // Find the target input via the selector in data-w-sync-target-value
+    const titleInput = document.querySelector<HTMLInputElement>(
+      this.targetValue,
+    );
+
+    if (!titleInput) {
+      return;
+    }
+
+    // Do not override if user has already typed a title
+    if (titleInput.value) return;
+
+    const rawPath = input.value || '';
+    const parts = rawPath.split('\\');
+    const filenameWithExt = parts[parts.length - 1] || '';
+
+    // Remove file extension
+    const title = filenameWithExt.replace(/\.[^.]+$/, '');
+
+    // Prepare details for the custom event
+    const maxTitleLengthAttr = titleInput.getAttribute('maxLength');
+    const maxTitleLength =
+      maxTitleLengthAttr !== null ? parseInt(maxTitleLengthAttr, 10) : null;
+
+    const customEvent = new CustomEvent('wagtail:images-upload', {
```

Comment:
We need to extract `wagtail:images-upload` into a Stimulus value or data attribute so we can use the same code for documents upload as well.

Although, if we use a Stimulus value, I think this might only be temporary. We won't need the value when we completely migrate all instances of this functionality to use the Stimulus implementation, as we can just use a generic event name e.g. `w-sync:upload` or something.

```suggestion
    const customEvent = new CustomEvent(this.eventValue, {
```